value/valuereflect_test.go

```diff
@@ -661,6 +661,33 @@ func TestReflectList(t *testing.T) {
 	}
 }
 
+func TestReflectListMutateSet(t *testing.T) {
```

Comment:
Can we have a test `Set`-ting something that is not a scalar? :-)

Reply:
added a test with two map types. one set using reflect which works. another is set using a value wrapping an unstructured, which doesnt